sqlobject/src/main/java/org/jdbi/v3/sqlobject/locator/internal/UseClasspathSqlLocatorImpl.java

```diff
@@ -15,31 +15,35 @@
 
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
+import java.util.Optional;
 
 import org.jdbi.v3.core.config.ConfigRegistry;
 import org.jdbi.v3.core.locator.ClasspathSqlLocator;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
 import org.jdbi.v3.sqlobject.SqlObjects;
 import org.jdbi.v3.sqlobject.config.Configurer;
 import org.jdbi.v3.sqlobject.internal.SqlAnnotations;
 import org.jdbi.v3.sqlobject.locator.SqlLocator;
 
 public class UseClasspathSqlLocatorImpl implements Configurer {
-    private static final SqlLocator SQL_LOCATOR = (sqlObjectType, method, config) -> {
-        return SqlAnnotations.getAnnotationValue(method,
-                name -> ClasspathSqlLocator.findSqlOnClasspath(sqlObjectType, defaultName(name, method))).orElseGet(method::getName);
-    };
+    private static final SqlLocator LOCATOR = (type, method, config) -> locate(type, method);
 
     @Override
     public void configureForType(ConfigRegistry registry, Annotation annotation, Class<?> sqlObjectType) {
-        registry.get(SqlObjects.class).setSqlLocator(SQL_LOCATOR);
-    }
-
-    private static String defaultName(String name, Method method) {
-        return name.isEmpty() ? method.getName() : name;
+        registry.get(SqlObjects.class).setSqlLocator(LOCATOR);
     }
 
     @Override
     public void configureForMethod(ConfigRegistry registry, Annotation annotation, Class<?> sqlObjectType, Method method) {
         configureForType(registry, annotation, sqlObjectType);
     }
+
+    private static String locate(Class<?> sqlObjectType, Method method) {
+        Optional<String> resolved = SqlAnnotations.getAnnotationValue(method, value -> {
+            String name = value.isEmpty() ? method.getName() : value;
+            return ClasspathSqlLocator.findSqlOnClasspath(sqlObjectType, name);
+        });
+
+        return resolved.orElseThrow(() -> new UnableToCreateStatementException(String.format("the query returned for %s was empty", method)));
```

Comment:
how about something more like `"unable to locate sql for " method` -- "the query returned" sounds like you've actually executed a query on the database, but we never actually did